observability-kit-test/observability-kit-agent-it/pom.xml

```diff
@@ -0,0 +1,174 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.vaadin</groupId>
+        <artifactId>observability-kit-test</artifactId>
+        <version>2.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>observability-kit-agent-it</artifactId>
+    <packaging>war</packaging>
+
+    <properties>
+        <failOnMissingWebXml>false</failOnMissingWebXml>
+        <otel.exporter>otlp</otel.exporter>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-server</artifactId>
+            <version>${flow.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>vaadin-dev</artifactId>
```

Comment:
is this a platform dependency?

Reply:
Not sure, it comes from the Start project and if I remove this the ITs won't run.

Reply:
you can replace it with separate dependencies. 
for example: 

```
        <dependency>
            <groupId>com.vaadin</groupId>
            <artifactId>vaadin-dev-server</artifactId>
        </dependency>
```
and probably some component dependencies.

Reply:
I pushed a commit to replace it.. let us see..

Reply:
Thanks! Is there a list of platform modules to check to avoid this to happen again? I mean, how do we know that `vaadin-dev` is a platform dependency and `vaadin-dev-server` is not? 🤔

Reply:
It seems most of the project POMs inherit from `vaadin-parent`. Does it make sense to add a profile there with a `bannedDependencies` enforcer rule that fails if there are platform dependencies in use?
Then we could also add a step in the validation job to perform the check and block at PR stage